Include/internal/pycore_frame.h

```diff
@@ -31,6 +31,7 @@ typedef struct _interpreter_frame {
     int f_lasti;       /* Last instruction if called */
     int stacktop;     /* Offset of TOS from localsplus  */
     PyFrameState f_state;  /* What state the frame is in */
+    int depth; /* Depth of the frame in a ceval loop */
```

Comment:
No, check this comment:

https://github.com/python/cpython/pull/28488#discussion_r712921230

This is here because using this information from debugging tools is much much easier this way. While I was fixing the gdb stuff, it became clear that having it into the cframe generates a considerable amount of complexity.

Reply:
I profiled this and it has no measurable impact on runtime performance.

Reply:
That's probably more a comment on our ability to measure small changes :disappointed: 
We can always change it later if it is slower.

What I'm more worried about is the redundancy and bugs due to out of date depth information on frames that outlive the call.

Reply:
What we could do is, instead of having a `depth` value, have an `entry` flag.
On entry to `_PyEval_EvalFrameDefault`, set the `entry` flag, and unset it on exit.
I think that would be safer, and no slower.

Reply:
> What I'm more worried about is the redundancy and bugs due to out of date depth information on frames that outlive the call.

As in generators or coroutines? Those will always have `depth=0`

Reply:
At the moment sure, but we will want to flatten `next/yield` in the same way as we are doing with `call/return` here.

Reply:
> but we will want to flatten `next/yield` in the same way as we are doing with `call/return` here.

But then it will be easy: on the exit path set it to `-1` or `0`.

Reply:
Can we change `depth` to a boolean, named e.g. `is_entry`. Two reasons:
* It is less misleading when the frame is not executing (e.g. when introspecting a frame).
* It would reduce the size of the frame (if not immediately due to layout, later when other fields are tidied up a bit).